File: easy/3945_digit_frequency_score/Solution.java
class Solution {
  public int digitFrequencyScore(int n) {
    int nCopy = n;
    int score = 0;

    while (nCopy != 0) {
      score += nCopy % 10;
      nCopy /= 10;
    }

    return score;
  }
}

File: easy/3945_digit_frequency_score/solution.py
class Solution:
    def digitFrequencyScore(self, n: int) -> int:
        frequencies = [0] * 10

        while n:
            frequencies[n % 10] += 1
            n //= 10

        score = 0

        for digit, frequency in enumerate(frequencies):
            score += digit * frequency

        return score
